src/sparseml/modifiers/smoothquant/base.py

```diff
@@ -111,13 +111,13 @@ def on_initialize(self, state: State, **kwargs) -> bool:
         """
         if self.end and self.end != -1:
             raise ValueError(
-                "SmoothQuantModifier can only be applied during one-shot. Expected end"
-                " to be None or -1, got {}".format(self.end)
+                f"{self.__class__.__name__} can only be applied during one-shot. "
+                f" Expected end to be None or -1, got {self.end}"
             )
         if self.start and self.start != -1:
```

Comment:
This condition evaluates to False if self.start == 0, which is probably not what you want.

Reply:
This guarantees it only applies as one-shot